ui/app/selectors/permissions.js

```diff
@@ -0,0 +1,65 @@
+
```

Comment:
All the selector stuff is probably worth paying extra attention to, because this was my first time using Reselect.